### PR TITLE
Lua blitbuffer support for CFA displays (Kaleido)

### PR DESCRIPTION
**Pull request by ezdiy**
Very experimental as of now, needs feedback from https://github.com/koreader/koreader/issues/6479

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1191)

### CODE REVIEW

File: ffi/blitbuffer.lua [hunk 1]
```diff
@@ -596,8 +596,10 @@ local MASK_INVERSE = 0x02
 local SHIFT_INVERSE = 1
 local MASK_ROTATED = 0x0C
 local SHIFT_ROTATED = 2
-local MASK_TYPE = 0xF0
+local MASK_TYPE = 0x70
```
Comment (NiLuJe) [inline ffi/blitbuffer.lua:599]: This'll need a matching change in blitbuffer.c ;).

**Comment by NiLuJe** [comment]
I'm not *quite* sure I grok everything happening in there, but, in any case, the comments are very welcome ^^.

(And, yeah, ultimately this should probably be implemented as a meta method once we have confirmation that it works).

**Comment by ezdiy** [comment]
> I'm not quite sure I grok everything happening in there

You can force your kobo to render CFA too just by giving it:

```
    canUseCBB = no,
    hasColorScreen = yes,
    hasCFAScreen = yes,
```

And then guesstimating under looking glass it indeed looks like the staircase patterns posted in koreader/koreader#6479. In the end the issues will be mostly rotation related (again!) because when the pattern can go BGR or RGB separately for each X or Y + starting offset when we're going backwards (display is even number pixels, but CFA is multiplle of 3...).

File: ffi/blitbuffer.lua (hunk 1 above)
Comment (ezdiy) [reply ffi/blitbuffer.lua:599]: Yep. Provided I'll get ad-hoc rotation going on correctly (ie no SW rotations ever), this is one of the places where C blitter can benefit a lot (the operation can be parallelized on neon for batches of 12 pixels).

**Comment by pazos** [comment]
Superseeded by #1451

**Comment by poire-z** [comment]
> Superseeded 

Until someday some other manufacturer like Kobo comes up with a device that uses one of these color eink panels :)

**Comment by NiLuJe** [comment]
Would probably be a non-issue on any other manufacturer, because in the updated sunxi drivers used on Kobo, enabling CFA support just takes toggling a flag in the ioctl struct, all the magic happen in kernel-land ;).

**Pull request closed without merging**